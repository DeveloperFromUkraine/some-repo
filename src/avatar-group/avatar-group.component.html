<div *ngIf="avatars?.length">
  <div id="avatar-group"
       fxLayout="row"
       tabindex="-1">
    <ign-avatar *ngFor="let avatar of sanitizeAvatars(avatars).slice(0, cap); let i = index"
                class="avatar avatar-group"
                matListAvatar
                [tooltip]="getAvatarTooltip(avatar)"
                [image]="avatar.image || ''"
                [initials]="avatar.person ? getPersonInitials(avatar.person) : ''"
                [size]="size"
                [routerLink]="avatar.link ? [avatar.link] : []"
                tabindex="-1"
                (click)="avatarClick.emit(avatar)"></ign-avatar>
    <ign-avatar *ngIf="showCounter && getCounterValue(avatars) > 0"
                id="counter"
                [class]="'avatar-group ' + (showContext ? 'counter--context-avatar' : 'counter--plus-avatar')"
                matListAvatar
                [initials]="getCounterInitials()"
                [size]="size"
                [routerLink]="counterLink ? [counterLink] : []"
                [ariaLabel]="counterAriaLabel || ' '"
                tabindex="-1"
                [attr.role]="counterLink ? 'link' : ''"
                (click)="counterClick.emit()"></ign-avatar>
  </div>
  <div *ngIf="showContext && viewAllLink">
        <button mat-button
                class="context__tool-all"
                [routerLink]="viewAllLink ? [viewAllLink] : []"
                attr.aria-label="{{ viewAllAriaLabel }}"
                (click)="viewAllClick.emit()">
                VIEW ALL ({{ this.getCounterValue(avatars) + this.cap }})
        </button>
  </div>
</div>
