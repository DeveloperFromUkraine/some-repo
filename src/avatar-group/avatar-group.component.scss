@import '~ignite-common/styles/ignite';

// Colors
$light-gray: ign-palette($ign-palette-gray, 150);
$dark-gray: ign-palette($ign-palette-gray, 500);

#avatar-group {
  display: flex;
}

.avatar-group {
  padding-top: 8px; // matches material design padding.
  margin-top: 0;
  margin-left: -8px;

  &:first-of-type {
    margin-left: 0;
  }
}

.context__all {
  padding-left: 0;
  padding-right: 0;
  color: ign-color(secondary);
  font-size: 0.875rem; // TODO: the typography font-size for button doesn't match spec, neither do raw tokens
  font-family: $ign-font-family;
  font-weight: 500; // TODO: the typography font-weight for buttons doesn't match spec, neither do raw tokens
}

:host ::ng-deep #counter {
  &.counter--plus {
    .container--initials {
      background-color: $light-gray;
    }

    .initials {
      font-size: ign-font-size(xxxs);
      color: $dark-gray;
      font-weight: ign-font-weight(regular);
      position: relative;
      left: -0.125rem; // assumes the size stays 'm'
    }
  }

  &.counter--context {
    .container--initials {
      background-color: $light-gray;
    }

    .initials {
      font-size: ign-font-size(xs);
      color: $dark-gray;
      font-weight: ign-font-weight(heavy);
      position: relative;
      top: -0.375rem; // assumes the size stays 'm'
    }
  }
}