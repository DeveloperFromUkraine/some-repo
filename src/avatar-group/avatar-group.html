<div *ngIf="avatars?.length"
     id="avatar-group"
     fxLayout="row"
     tabindex="-1">
    <ign-avatar *ngFor="let avatar of sanitizeAvatars(avatars).slice(0, cap)"
                class="avatar avatar-group"
                matListAvatar
                [tooltip]="getAvatarTooltip(avatar)"
                [image]="avatar.image || ''"
                [initials]="avatar.person ? getPersonInitials(avatar.person) : ''"
                [size]="size"
                [routerLink]="avatar.link ? [avatar.link] : []"
                tabindex="-1"></ign-avatar>
    <ign-avatar *ngIf="!!showCounter && getCounterValue(avatars) > 0"
                id="counter"
                class="avatar-group"
                matListAvatar
                [initials]="'+' + getCounterValue(avatars)"
                [size]="size"
                [routerLink]="showCounter && counterLink ? [counterLink] : []"
                [ariaLabel]="counterAriaLabel || ' '"
                tabindex="-1"
                [attr.role]="showCounter && counterLink ? 'link' : ''"></ign-avatar>
</div>