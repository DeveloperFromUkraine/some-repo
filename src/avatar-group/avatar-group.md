*** HTML ***

# Avatar Group

## Use

The avatar group component horizontally displays several avatars and, optionally, a count of avatars not displayed.

Each avatar has a configurable tooltip that renders either a standardized full name (if a person is provided), or custom text (if a tooltip value is provided).

Each avatar, and the counter, can be configured to contain a url link to be navigated to on click.

The maximum number of avatars displayed, and not displayed (the value seen in the counter), can be configured.

Type `Person` has been added to facilitate the casting of other types to meet interface Avatar's attribute `person` (see sample usage).
## Sample

<mat-tab-group>
    <mat-tab label="Component Sample">
        <div class="tab-height">
          <ign-avatar-group
            [avatars]="avatars"
            [cap]="avatarsCap"
            [counterAriaLabel]="avatars.length - cap + counterOffset + ' more employees available in the Org Chart'"
            [counterOffset]="counterOffset"
          ></ign-avatar-group>
        </div></mat-tab>
    <mat-tab label="HTML"><div class="tab-height">
        <table style="width:100%">
            <p>&lt;ign-avatar-group&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;[avatars]="avatars"</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;[cap]="avatarsCap"</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;[counterAriaLabel]="avatars.length - cap + counterOffset + ' more employees available in the Org Chart'"</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;[counterOffset]="counterOffset"</p>
            <p>&lt;/ign-avatar-group&gt;</p>
        </table>
    </div></mat-tab>
    <mat-tab label="TS"><div class="tab-height">
        <table style="width:100%">
            <p>// This is just an example avatar group data source.</p>
            <br>
            <p>// Visualize this as a type that contains attribute employeeInfo, a fictitious superset of type Person.</p>
            <p>@Input data: EmployeeData[];</p>
            <br>
            <p>avatars: Avatar[];</p>
            <p>avatarsCap = 3;</p>
            <br>
            <p>ngOnChanges(changes: SimpleChanges) &#123;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;const dataChanges = changes['data'];</p>
            <br>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;if (dataChanges && data.length) &#123;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;this.avatars = this.generateAvatars(this.data);</p>
            <p>&nbsp;&nbsp;&#125;</p>
            <p>&#125;</p>
            <br>
            <p>generateAvatars(data: EmployeeData[]): Avatar[] &#123;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;let avatars: Avatar[] = [];</p>
            <br>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;for (const dataItem of employeeData) &#123;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;const avatar: Avatar = &#123;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;person: <Person>dataItem.employeeInfo,</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;link: dataItem.employeeInfo</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;? `/employees/$&#123;data.dataItem.employmentId&#125;/employment`</p>      
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;: `/activations/position-hub/positions/$&#123;dataItem.position.positionId&#125;`,</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&#125;</p>
            <br>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;avatars = [...avatars, avatar];</p>
            <p>&nbsp;&nbsp;&#125;</p>
            <p>&#125;</p>
        </table>
    </div></mat-tab>
</mat-tab-group>

## Content

* `@Input() avatars: Avatar[]`
  * The avatars to be displayed, where:
  ```
    export interface Avatar {
      person?: Person;
      image?: string;
      // Use one string with a relative path. Do not use a route parameter array.
      link?: string;
      tooltip?: string;
    }

    export type Person = {
      firstName: string;
      lastName: string;
      preferredFirstName?: string;
      formerLastName?: string;
    }
  ```

* `@Input() cap: number`
  * Optional input. By default, cap = 8.
  * The maximum number of avatars to be rendered, regardless of length of Avatar[].
  
* `@Input() size: string`
  * The size of the avatar. Value can be 'small', or 'large'.

* `@Input() showCounter: boolean`
  * Optional input. Default value is true.
  * Whether to show the counter of the number of avatars not being rendered.
  * The counter will only be displayed if the number of avatars is greater than the value of cap.

* `@Input() counterOffset: number`
  * Optional input. Default value is 0.
  * The counter offset is a number added to the avatars's counter in order to modify its rendered value 
  * Example: Each avatar represents an employee and a company has 30 employees; however, you only have data to generate 10 avatars. 
  * If a cap has not been specified, the avatar group component will generate 10 avatars, render 8, and display a counter with value 2. If you want this counter to represent the total amount of employees within the company, you could set counterOffset to equal `totalNumberOfEmployees - avatars.length`.

* `@Input() counterLink: string`
  * Optional input.
  * When a user clicks on the counter, the router navigates to the specified URI.
  * Use one string with a relative path. Do not use a routes parameter array.

* `@Input() counterAriaLabel: string`
  * Optional input.
  * The value to set the counter's attribute `aria-label` to.

## Accessibility

* The counter, by default, has attribute `tabindex` set to `"-1"`.
  * Providing a value for `counterLink` changes the counter's attrinute `tabindex` to `"0"`.
  * Providing a value for `counterAriaLabel` changes the counter's attrinute `tabindex` to `"0"`.
* Each avatar's attribute `aria-label` is set to the value of its tooltip.

## Style

* The avatar group is already pre-styled.

*** TypeScript ***

  @Input() avatars: Avatar[] = [
    {
      person: {
        firstName: 'firstName1',
        preferredFirstName: 'preferredFirstName1',
        lastName: 'lastName1',
        formerLastName: 'formerLastName1',
      },
      tooltip: 'tooltip1',
    },
    {
      person: {
        firstName: 'firstName2',
        preferredFirstName: 'preferredFirstName2',
        lastName: 'lastName2',
        formerLastName: 'formerLastName2',
      },
      image: 'photoUploadId2',
      tooltip: 'tooltip2',
    },
    {
      person: {
        firstName: 'firstName3',
        preferredFirstName: 'preferredFirstName3',
        lastName: 'lastName3',
        formerLastName: 'formerLastName3',
      },
      image: 'photoUploadId3',
      tooltip: 'tooltip3',
    }
  ];

  avatarsCap = 3;
  counterOffset = 2;

  size = 'small';

*** ExternalTypeScript ***

*** Imports ***

import { Component, Input } from '@angular/core';
