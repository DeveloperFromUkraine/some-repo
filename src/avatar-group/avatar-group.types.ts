export interface Avatar {
  person?: Person;
  image?: string;
  link?: string;
  tooltip?: string;
}

export type Person = {
  firstName: string;
  lastName: string;
  preferredFirstName?: string;
  formerLastName?: string;
}
