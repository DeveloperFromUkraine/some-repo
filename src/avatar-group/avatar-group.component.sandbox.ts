import { sandboxOf } from 'angular-playground';
import { AvatarGroupComponent } from './avatar-group.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { MatTooltipModule, MatIconModule } from '@angular/material';
import { AvatarComponent } from '../avatar/avatar.component';

import { RouterModule, Routes } from '@angular/router';

import { Avatar } from './avatar-group.types';

const appRoutes: Routes = [{ path: 'UI/part1/Details', component: AvatarComponent }];

let avatars: Avatar[] = [
  {
    person: {
      firstName: 'Scott',
      lastName: 'Brooks',
      preferredFirstName: 'Scottie B',
    },
    image: 'http://i.pravatar.cc/150?img=6',
    tooltip: 'Ahhh!',
  },
  {
    person: {
      firstName: 'Michelle',
      lastName: 'Beadle',
      preferredFirstName: 'Melly Mel',
    },
    image: 'http://i.pravatar.cc/150?img=5',
  },
  {
    person: {
      firstName: 'Main',
      lastName: 'Man',
      preferredFirstName: 'Da Boss',
    },
    image: 'http://i.pravatar.cc/150?img=54',
    tooltip: 'Ahhh!',
  },
  {
    person: {
      firstName: 'Shy',
      lastName: 'Guy',
    },
  },
];

export default sandboxOf(AvatarGroupComponent, {
  imports: [
    MatTooltipModule,
    MatIconModule,
    RouterModule.forRoot(appRoutes),
    BrowserAnimationsModule,
  ],
  declarations: [AvatarComponent],
}).add('Display 4 avatars in a group: 3 with images 1 with initials', {
  template: `
    <ign-avatar-group
      [avatars]="avatars"
      [cap]="5"
      [counterAriaLabel]="'2 more in the org chart.'">
    </ign-avatar-group>
`,
  context: {
    avatars,
  },
});
