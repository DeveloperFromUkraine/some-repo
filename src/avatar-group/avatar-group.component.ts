import { Component, Input, Output, EventEmitter } from '@angular/core';
import { Avatar, Person } from './avatar-group.types';

@Component({
  selector: 'ign-avatar-group',
  templateUrl: './avatar-group.component.html',
  styleUrls: ['./avatar-group.component.scss'],
})
export class AvatarGroupComponent {
  @Input() avatars: Avatar[];
  @Input() cap = 8;
  @Input() showCounter = true;
  @Input() showContext = false;
  @Input() counterOffset = 0;
  @Input() counterLink: string;
  @Input() counterAriaLabel: string;
  @Input() viewAllLink: string;
  @Input() viewAllAriaLabel: string;

  @Output() avatarClick = new EventEmitter<Avatar>();
  @Output() counterClick = new EventEmitter();
  @Output() viewAllClick = new EventEmitter();

  // Set size of all avatars in the group to medium
  size = 'm';

  getAvatarTooltip(avatar: Avatar): string {
    if (avatar.tooltip) {
      return avatar.tooltip;
    }
    return avatar.person ? this.getPersonName(avatar.person) : '';
  }

  getCounterInitials(): string {
    return this.showContext ? '...' : `+${this.getCounterValue(this.avatars)}`;
  }

  getCounterValue(avatars: Avatar[]): number {
    return avatars.length - this.cap + this.counterOffset;
  }

  getPersonInitials(person: Person): string {
    const first = person.preferredFirstName || person.firstName;
    const initials = first.charAt(0) + person.lastName.charAt(0);
    return initials.toUpperCase();
  }

  getPersonName(person: Person): string {
    const first = person.preferredFirstName || person.firstName;
    const former = !person.formerLastName ? '' : `(${person.formerLastName})`;
    return this.sanitizePersonName(`${first} ${person.lastName} ${former}`).trim();
  }

  /**
   * This method prevents attempts to create avatars with invalid data. It is
   * not meant to prevent compilation-time errors, since interface Avatar already
   * accomplishes that, but rather prevent runtime errors.
   * @param avatars
   */
  sanitizeAvatars(avatars: Avatar[]): Avatar[] {
    avatars = avatars.filter(avatar => {
      const isEmployeeValid = avatar.person && avatar.person.lastName && avatar.person.firstName;
      return avatar.person ? isEmployeeValid : !!avatar.tooltip;
    });
    return avatars;
  }

  sanitizePersonName(fullName: string): string {
    let names = fullName.split(' ');
    names = names.filter(name => !!name && name !== 'undefined');
    return names.join(' ');
  }
}
