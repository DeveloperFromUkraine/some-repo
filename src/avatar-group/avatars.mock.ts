import { Avatar } from './avatar-group.types';

// All avatars' properties are present and valid.
export const avatars4: Avatar[] = [
  {
    person: {
      firstName: 'firstName1',
      preferredFirstName: 'preferredFirstName1',
      lastName: 'lastName1',
      formerLastName: 'formerLastName1',
    },
    link: 'redirectUrl1',
    tooltip: 'tooltip1',
  },
  {
    person: {
      firstName: 'firstName2',
      preferredFirstName: 'preferredFirstName2',
      lastName: 'lastName2',
      formerLastName: 'formerLastName2',
    },
    link: 'redirectUrl2',
    image: 'photoUploadId2',
    tooltip: 'tooltip2',
  },
  {
    person: {
      firstName: 'firstName3',
      preferredFirstName: 'preferredFirstName3',
      lastName: 'lastName3',
      formerLastName: 'formerLastName3',
    },
    image: 'photoUploadId3',
    link: 'redirectUrl3',
    tooltip: 'tooltip3',
  },
];

// Each avatar requires a different tooltip computation.
export const avatars5: Avatar[] = [
  // Has explicit tooltip, which should override any computation.
  {
    person: {
      firstName: 'firstName1',
      preferredFirstName: 'preferredFirstName1',
      lastName: 'lastName1',
      formerLastName: 'formerLastName1',
    },
    image: 'photoUploadId1',
    link: 'redirectUrl1',
    tooltip: 'tooltip1',
  },
  // Has empty tooltip, but person fullname can be computed.
  {
    person: {
      firstName: 'firstName2',
      preferredFirstName: 'preferredFirstName2',
      lastName: 'lastName2',
      formerLastName: 'formerLastName2',
    },
    link: 'redirectUrl2',
    image: 'photoUploadId2',
    tooltip: '',
  },
  // Does not have tooltip, but person fullname can be computed.
  {
    person: {
      firstName: 'firstName3',
      lastName: 'lastName3',
    },
    image: 'photoUploadId3',
    link: 'redirectUrl3',
  },
  // Does not have tooltip, but person fullname can be computed.
  {
    person: {
      firstName: 'firstName4',
      lastName: 'lastName4',
      preferredFirstName: 'preferredFirstName4',
    },
    image: 'photoUploadId4',
    link: 'redirectUrl4',
  },
  // Does not have tooltip, but person fullname can be computed.
  {
    person: {
      firstName: 'firstName5',
      lastName: 'lastName5',
      formerLastName: 'formerLastName5',
    },
    image: 'photoUploadId5',
    link: 'redirectUrl5',
  },
];

/*
 * These are corner case avatars. Some contain missing fields necessary for some
 * tooltip computations while containing all fields necessary for the prioritized
 * tooltip computation. Other avatars are invalid and should be gracefully error
 * handled.
 */
export const avatarsX: Avatar[] = [
  // Valid Avatar. All fields are present and valid.
  {
    person: {
      firstName: 'firstName1',
      preferredFirstName: 'preferredFirstName1',
      lastName: 'lastName1',
      formerLastName: 'formerLastName1',
    },
    image: 'photoUploadId1',
    link: 'redirectUrl1',
  },
  // Invalid avatar: missing person first name.
  {
    person: {
      firstName: null,
      preferredFirstName: 'preferredFirstName2',
      lastName: 'lastName2',
      formerLastName: 'formerLastName2',
    },
    image: 'photoUploadId2',
    link: 'redirectUrl2',
  },
  // Valid avatar, but missing person preferred first name
  {
    person: {
      firstName: 'firstName3',
      lastName: 'lastName3',
      formerLastName: 'formerLastName3',
    },
    image: 'photoUploadId3',
    link: 'redirectUrl3',
  },
  // Invalid avatar: missing person last name.
  {
    person: {
      firstName: 'firstName4',
      lastName: null,
      preferredFirstName: 'preferredFirstName4',
      formerLastName: 'formerLastName4',
    },
    image: 'photoUploadId4',
    link: 'redirectUrl4',
  },
  // Valid avatar, but missing person former last name.
  {
    person: {
      firstName: 'firstName5',
      preferredFirstName: 'preferredFirstName5',
      lastName: 'lastName5',
    },
    link: 'redirectUrl5',
  },
  // Invalid avatar: missing person first and preferred names.
  {
    person: {
      firstName: null,
      preferredFirstName: null,
      lastName: 'lastName7',
      formerLastName: 'formerLastName7',
    },
    image: 'photoUploadId7',
    link: 'redirectUrl7',
  },
  // Invalid avatar: missing person former and last name.
  {
    person: {
      firstName: 'firstName8',
      lastName: null,
      formerLastName: null,
      preferredFirstName: 'preferredFirstName8',
    },
    image: 'photoUploadId8',
    link: 'redirectUrl8',
  },
  // Invalid avatar: missing person and tooltip.
  {
    image: 'redirectUrl9',
  },
  // Valid avatar: missing all fields other than tooltip.
  {
    tooltip: 'tooltip10',
  },
];
