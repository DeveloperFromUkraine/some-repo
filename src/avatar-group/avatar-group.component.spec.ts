import { DebugElement } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { By } from '@angular/platform-browser';
import { MatTooltipModule, MatIconModule } from '@angular/material';

import { AvatarGroupComponent } from '../avatar-group/avatar-group.component';
import { AvatarComponent } from '../avatar/avatar.component';
import { ComponentTest } from '../../test/test-bed/component';
import { avatars4 } from './avatars.mock';

describe('AvatarGroupComponent Test Suite', () => {
  let fixture: ComponentFixture<AvatarGroupComponent>;
  let component: AvatarGroupComponent;
  let de: DebugElement;

  beforeEach(async () => {
    await ComponentTest.createTestBed(
      [RouterTestingModule, MatTooltipModule, MatIconModule],
      [AvatarComponent, AvatarGroupComponent]
    );
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AvatarGroupComponent);
    component = fixture.componentInstance;
    de = fixture.debugElement;
  });

  it('should not display the avatars', async () => {
    let avatarGroup: DebugElement;

    fixture.detectChanges();
    avatarGroup = de.query(By.css('#avatar-group'));

    expect(avatarGroup).toBeNull();
  });

  it('should display the avatars', () => {
    component.avatars = avatars4;
    component.cap = 3;
    let displayedAvatars;

    fixture.detectChanges();
    displayedAvatars = de.queryAll(By.css('.avatar'));

    expect(displayedAvatars.length).toBe(3);
  });

  it('should not display more avatars than the cap number', () => {
    component.avatars = avatars4;
    component.cap = 2;
    let avatars: Array<DebugElement>;

    fixture.detectChanges();
    avatars = de.queryAll(By.css('.avatar'));

    expect(avatars.length).toBe(2);
  });

  it('should have counter attribute "aria-label" set to counterAriaLabel', () => {
    component.avatars = avatars4;
    component.cap = 2;
    component.counterAriaLabel = '4 More in the Org Chart';
    let counter: DebugElement;
    let ne: HTMLElement;

    fixture.detectChanges();
    counter = de.query(By.css('#counter'));
    ne = counter.nativeElement;

    expect(ne.getAttribute('ng-reflect-aria-label')).toContain('4 More in the Org Chart');
  });

  it('should have counter attribute "routerLink" set to counterLink', () => {
    component.avatars = avatars4;
    component.cap = 2;
    component.counterLink = 'counterRedirectUrl';
    let counter: DebugElement;
    let ne: HTMLElement;

    fixture.detectChanges();
    counter = de.query(By.css('#counter'));
    ne = counter.nativeElement;

    expect(ne.getAttribute('ng-reflect-router-link')).toContain('counterRedirectUrl');
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
