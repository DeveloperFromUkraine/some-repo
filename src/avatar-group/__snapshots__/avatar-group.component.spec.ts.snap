// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroupComponent Test Suite should match snapshot 1`] = `
<ign-avatar-group
  avatarClick={[Function EventEmitter]}
  cap={[Function Number]}
  counterClick={[Function EventEmitter]}
  counterOffset="0"
  showContext="false"
  showCounter={[Function Boolean]}
  showTooltips={[Function Boolean]}
  size={[Function String]}
  viewAllClick={[Function EventEmitter]}
>
  
</ign-avatar-group>
`;
